File: 141.php
<?php
/**
 * Definition for a singly-linked list.
 * class ListNode {
 *     public $val = 0;
 *     public $next = null;
 *     function __construct($val) { $this->val = $val; }
 * }
 */

class Solution {
  /**
   * @param ListNode $head
   * @return Boolean
   */
  function hasCycle($head) {
    if (!$head || !$head->next) {
      return false;
    }
    $fast = $head->next;
    $slow = $head;
    while ($fast) {

      if ($fast === $slow) {
        return true;
      }
      $fast = $fast->next->next;
      $slow = $slow->next;
    }
    return false;
  }
}

File: go/linkedList/141.go
package main

//func hasCycle(head *ListNode) bool {
//	//flag := make(map[*ListNode]bool)
//	//
//	//for head != nil {
//	//	if _, ok := flag[head]; ok {
//	//		return true
//	//	}
//	//	flag[head] = true
//	//	head = head.Next
//	//}
//	//return false
//
//	fast, slow := head, head
//
//	for fast != nil && slow != nil && fast.Next != nil {
//		fast = fast.Next.Next
//		slow = slow.Next
//
//		if fast == slow {
//			return true
//		}
//	}
//	return false
//}
//
//type ListNode struct {
//	Val  int
//	Next *ListNode
//}
